src/modules/hip/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,2065 @@
+    dst_f8->f1[4] = fmaf(src_f1, filter_f9->f1[rowIndex + 2], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter_f9->f1[rowIndex + 1], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter_f9->f1[rowIndex], dst_f8->f1[6]);
+    src_f1 = rpp_hip_unpack3(src_ui3.y);
+    dst_f8->f1[5] = fmaf(src_f1, filter_f9->f1[rowIndex + 2], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter_f9->f1[rowIndex + 1], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter_f9->f1[rowIndex], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack0(src_ui3.z);
+    dst_f8->f1[6] = fmaf(src_f1, filter_f9->f1[rowIndex + 2], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter_f9->f1[rowIndex + 1], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack1(src_ui3.z);
+    dst_f8->f1[7] = fmaf(src_f1, filter_f9->f1[rowIndex + 2], dst_f8->f1[7]);
+}
+
+__device__ void gaussian_filter_5x5_row_hip_compute(uchar *srcPtr, d_float8 *dst_f8, d_float25 *filter_f25, int rowIndex)
+{
+    float src_f1;
+    uint3 src_ui3;
+    src_ui3 = *(uint3 *)srcPtr;
+    src_f1 = rpp_hip_unpack0(src_ui3.x);
+    dst_f8->f1[0] = fmaf(src_f1, filter_f25->f1[rowIndex], dst_f8->f1[0]);
+    src_f1 = rpp_hip_unpack1(src_ui3.x);
+    dst_f8->f1[0] = fmaf(src_f1, filter_f25->f1[rowIndex + 1], dst_f8->f1[0]);
+    dst_f8->f1[1] = fmaf(src_f1, filter_f25->f1[rowIndex], dst_f8->f1[1]);
+    src_f1 = rpp_hip_unpack2(src_ui3.x);
+    dst_f8->f1[0] = fmaf(src_f1, filter_f25->f1[rowIndex + 2], dst_f8->f1[0]);
+    dst_f8->f1[1] = fmaf(src_f1, filter_f25->f1[rowIndex + 1], dst_f8->f1[1]);
+    dst_f8->f1[2] = fmaf(src_f1, filter_f25->f1[rowIndex], dst_f8->f1[2]);
+    src_f1 = rpp_hip_unpack3(src_ui3.x);
+    dst_f8->f1[0] = fmaf(src_f1, filter_f25->f1[rowIndex + 3], dst_f8->f1[0]);
+    dst_f8->f1[1] = fmaf(src_f1, filter_f25->f1[rowIndex + 2], dst_f8->f1[1]);
+    dst_f8->f1[2] = fmaf(src_f1, filter_f25->f1[rowIndex + 1], dst_f8->f1[2]);
+    dst_f8->f1[3] = fmaf(src_f1, filter_f25->f1[rowIndex], dst_f8->f1[3]);
+    src_f1 = rpp_hip_unpack0(src_ui3.y);
+    dst_f8->f1[0] = fmaf(src_f1, filter_f25->f1[rowIndex + 4], dst_f8->f1[0]);
+    dst_f8->f1[1] = fmaf(src_f1, filter_f25->f1[rowIndex + 3], dst_f8->f1[1]);
+    dst_f8->f1[2] = fmaf(src_f1, filter_f25->f1[rowIndex + 2], dst_f8->f1[2]);
+    dst_f8->f1[3] = fmaf(src_f1, filter_f25->f1[rowIndex + 1], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter_f25->f1[rowIndex], dst_f8->f1[4]);
+    src_f1 = rpp_hip_unpack1(src_ui3.y);
+    dst_f8->f1[1] = fmaf(src_f1, filter_f25->f1[rowIndex + 4], dst_f8->f1[1]);
+    dst_f8->f1[2] = fmaf(src_f1, filter_f25->f1[rowIndex + 3], dst_f8->f1[2]);
+    dst_f8->f1[3] = fmaf(src_f1, filter_f25->f1[rowIndex + 2], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter_f25->f1[rowIndex + 1], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter_f25->f1[rowIndex], dst_f8->f1[5]);
+    src_f1 = rpp_hip_unpack2(src_ui3.y);
+    dst_f8->f1[2] = fmaf(src_f1, filter_f25->f1[rowIndex + 4], dst_f8->f1[2]);
```

Comment:
Please remove all these adds

Reply:
Done this for 3x3 kernels. Need to replicate for other kernel sizes
https://github.com/sampath1117/rpp/blob/sr/opt_gauss_filter_hip/src/modules/hip/kernel/gaussian_filter.hpp#L6
https://github.com/sampath1117/rpp/blob/sr/opt_gauss_filter_hip/src/modules/hip/kernel/gaussian_filter.hpp#L302
https://github.com/sampath1117/rpp/blob/sr/opt_gauss_filter_hip/src/modules/hip/kernel/gaussian_filter.hpp#L339

Reply:
Replicated for remaining kernel sizes